lib/ansible/modules/cloud/vmware/vmware_content_library_manager.py

```diff
@@ -0,0 +1,261 @@
+- All variables and VMware object names are case sensitive.
+version_added: '2.9'
+author:
+- Pavan Bidkar (@pgbidkar)
+notes:
+- Tested on vSphere 6.5, 6.7
+requirements:
+- python >= 2.6
+- PyVmomi
+- vSphere Automation SDK
+options:
+    library_name:
+      description:
+      - The name of library_name to create.
+      type: str
+      required: True
+    library_description:
+      description:
+      - The content library description .
+      - This is required only if C(state) is set to C(present).
+      - This parameter is ignored, when C(state) is set to C(absent).
+      - Process of updating content library only allows description change.
+      type: str
+      required: False
+      default: ''
+    datastore_name:
+      description:
+      - Name of the datastore on which backing content library is created
+      - This is required only if C(state) is set to C(present).
+      - This parameter is ignored, when C(state) is set to C(absent).
+      - Currently only datastore backing creation is supported
+      type: str
+      required: False
+    state:
+      description:
+      - The state of content library.
+      - If set to C(present) and library does not exists, then tag is created.
+      - If set to C(present) and library exists, then tag is updated.
+      - If set to C(absent) and library exists, then tag is deleted.
+      - If set to C(absent) and library does not exists, no action is taken.
+      type: str
+      required: False
+      default: 'present'
+      choices: [ 'present', 'absent' ]
+extends_documentation_fragment: vmware_rest_client.documentation
+'''
+
+EXAMPLES = r'''
+- name: Create Content Library
+  vmware_content_library_manager:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    library_name: test-content-lib
+    library_description: 'Library with Datastore Backing'
+    datastore_name: datastore
+    validate_certs: False
+    state: present
+  delegate_to: localhost
+
+- name: Update Content Library
+  vmware_content_library_manager:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    library_name: test-content-lib
+    library_description: 'Library with Datastore Backing'
+    validate_certs: no
+    state: present
+  delegate_to: localhost
+
+- name: Delete Content Library
+  vmware_content_library_manager:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    library_name: test-content-lib
+    validate_certs: no
+    state: absent
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+results:
+  description: libray creation success and library_id
+  returned: on success
+  type: dict
+  sample: {
+      "library_id": "d0b92fa9-7039-4f29-8e9c-0debfcb22b72",
+      "msg": "Content Library 'demo-local-lib-4' created."
```

Comment:
It would be nice to have other information about library like description and type etc.

Reply:
Sure. Added more details